frame/staking/src/weights.rs

```diff
@@ -77,192 +76,178 @@ pub trait WeightInfo {
 			.saturating_add(T::DbWeight::get().writes((1 as Weight).saturating_mul(s as Weight)))
 	}
 	fn new_era(v: u32, n: u32, ) -> Weight {
 		(0 as Weight)
-			// Standard Error: 903_000
-			.saturating_add((594_145_000 as Weight).saturating_mul(v as Weight))
-			// Standard Error: 45_000
-			.saturating_add((83_373_000 as Weight).saturating_mul(n as Weight))
-			.saturating_add(T::DbWeight::get().reads(9 as Weight))
-			.saturating_add(T::DbWeight::get().reads((4 as Weight).saturating_mul(v as Weight)))
+			// Standard Error: 1_013_000
+			.saturating_add((382_529_000 as Weight).saturating_mul(v as Weight))
+			// Standard Error: 50_000
+			.saturating_add((63_170_000 as Weight).saturating_mul(n as Weight))
+			.saturating_add(T::DbWeight::get().reads(10 as Weight))
+			.saturating_add(T::DbWeight::get().reads((3 as Weight).saturating_mul(v as Weight)))
 			.saturating_add(T::DbWeight::get().reads((3 as Weight).saturating_mul(n as Weight)))
-			.saturating_add(T::DbWeight::get().writes(13 as Weight))
+			.saturating_add(T::DbWeight::get().writes(9 as Weight))
 			.saturating_add(T::DbWeight::get().writes((3 as Weight).saturating_mul(v as Weight)))
 	}
-	fn submit_solution_better(v: u32, n: u32, a: u32, w: u32, ) -> Weight {
-		(0 as Weight)
-			// Standard Error: 52_000
-			.saturating_add((1_460_000 as Weight).saturating_mul(v as Weight))
-			// Standard Error: 20_000
-			.saturating_add((754_000 as Weight).saturating_mul(n as Weight))
-			// Standard Error: 52_000
-			.saturating_add((74_798_000 as Weight).saturating_mul(a as Weight))
-			// Standard Error: 108_000
-			.saturating_add((8_108_000 as Weight).saturating_mul(w as Weight))
-			.saturating_add(T::DbWeight::get().reads(6 as Weight))
-			.saturating_add(T::DbWeight::get().reads((4 as Weight).saturating_mul(a as Weight)))
-			.saturating_add(T::DbWeight::get().reads((1 as Weight).saturating_mul(w as Weight)))
-			.saturating_add(T::DbWeight::get().writes(2 as Weight))
-	}
 	fn get_npos_voters(v: u32, n: u32, s: u32, ) -> Weight {
 		(0 as Weight)
-			// Standard Error: 94_000
-			.saturating_add((29_321_000 as Weight).saturating_mul(v as Weight))
-			// Standard Error: 94_000
-			.saturating_add((66_885_000 as Weight).saturating_mul(n as Weight))
-			// Standard Error: 1_283_000
-			.saturating_add((22_991_000 as Weight).saturating_mul(s as Weight))
-			.saturating_add(T::DbWeight::get().reads(2 as Weight))
-			.saturating_add(T::DbWeight::get().reads((4 as Weight).saturating_mul(v as Weight)))
+			// Standard Error: 90_000
+			.saturating_add((27_108_000 as Weight).saturating_mul(v as Weight))
+			// Standard Error: 90_000
+			.saturating_add((29_962_000 as Weight).saturating_mul(n as Weight))
+			// Standard Error: 1_228_000
+			.saturating_add((26_080_000 as Weight).saturating_mul(s as Weight))
+			.saturating_add(T::DbWeight::get().reads(3 as Weight))
+			.saturating_add(T::DbWeight::get().reads((3 as Weight).saturating_mul(v as Weight)))
```

Comment:
@thiolliere I applied your suggestion for BTreeMap, the numbers don't change that much, but the DB weight is now clearer. Previously, I don't know why but the benchmark tool would mix the slashing spans read into `4 * v`, now we have `3 * v + 1 * s`, which is more sensible.